### PR TITLE
add more unit test in status handler to increase code coverage

### PR DESCRIPTION
**Pull request by feng-j678**
* add more unit tests to increase status handler code coverage by .89%
* current status handler coverage:
image
* new status handler coverage:
image

### CODE REVIEW

File: src/core/tests/Test_StatusHandler.py [hunk 1]
```diff
@@ -401,6 +401,47 @@ def test_sort_packages_by_classification_and_state(self):
...
+        self.runtime.status_handler.set_current_operation(Constants.INSTALLATION)
+
+        self.runtime.status_handler.report_sequence_number_changed_termination()
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][0]
+            self.assertTrue(substatus_file_data["name"] == Constants.PATCH_INSTALLATION_SUMMARY)
+            self.assertEqual(substatus_file_data["status"], Constants.STATUS_ERROR.lower())
+            formatted_message = json.loads(substatus_file_data['formattedMessage']['message'])
+            self.assertTrue(formatted_message["errors"]["details"][0]["code"] == Constants.PatchOperationErrorCodes.NEWER_OPERATION_SUPERSEDED)
+
```
Comment (kjohn-msft) [inline src/core/tests/Test_StatusHandler.py:444]: For all tests, below the definition describe what the test is validating (both to capture test author expectations and to allow for faster 'reads' of code in the future).

File: src/core/tests/Test_StatusHandler.py [hunk 2]
```diff
@@ -401,6 +401,47 @@ def test_sort_packages_by_classification_and_state(self):
...
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["name"], "python-samba")
+        self.assertTrue("Other" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["classifications"]))
+        self.assertEqual("Available", str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["patchInstallationState"]))
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][1]["name"], "samba-common-bin")
+        self.assertTrue("Other" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][1]["classifications"]))
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][2]["name"], "samba-libs")
+        self.assertTrue("python-samba_2:4.4.5+dfsg-2ubuntu5.4" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["patchId"]))
+        self.assertTrue("Other" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][2]["classifications"]))
+
+    def test_sequence_number_changed_termination_configuration_only(self):
```
Comment (kjohn-msft) [inline src/core/tests/Test_StatusHandler.py:421]: configuration != configure_patching   (use standard terminology)

File: src/core/tests/Test_StatusHandler.py [hunk 3]
```diff
@@ -401,6 +401,47 @@ def test_sort_packages_by_classification_and_state(self):
...
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["name"], "python-samba")
+        self.assertTrue("Other" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["classifications"]))
+        self.assertEqual("Available", str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["patchInstallationState"]))
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][1]["name"], "samba-common-bin")
+        self.assertTrue("Other" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][1]["classifications"]))
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][2]["name"], "samba-libs")
+        self.assertTrue("python-samba_2:4.4.5+dfsg-2ubuntu5.4" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][0]["patchId"]))
+        self.assertTrue("Other" in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][2]["classifications"]))
```
Comment (kjohn-msft) [inline src/core/tests/Test_StatusHandler.py:419]: What is this verifying that's not verified in some other test? Available? Available is not supposed to ever show up for patch installation - there's no actual code path that uses this in installation. 

If anything, the code is supposed to reject the set in set_package_install_status if some other future product code mistakenly tries to do this.

**Review by kjohn-msft — COMMENTED** [review]
Comments inline

File: src/core/tests/Test_StatusHandler.py (hunk 1 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py]: done

File: src/core/tests/Test_StatusHandler.py (hunk 2 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py]: done

File: src/core/tests/Test_StatusHandler.py (hunk 3 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py]: test removed, it's redundant test.

**Review by kjohn-msft — APPROVED** [review]

File: src/core/tests/Test_StatusHandler.py [hunk 4]
```diff
@@ -568,6 +576,19 @@ def test_remove_old_complete_status_files_throws_exception(self):
...
 
+    def __assert_sequence_num_changed_termination(self, config, summary, status):
+        self.runtime.execution_config.operation = config
+        self.runtime.status_handler.set_current_operation(config)
+
+        self.runtime.status_handler.report_sequence_number_changed_termination()
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][0]
+            self.assertTrue(substatus_file_data["name"] == summary)
+            self.assertTrue(substatus_file_data["name"] == summary)
```
Comment (rane-rajasi) [inline src/core/tests/Test_StatusHandler.py:587]: Duplicate lines

**Review by rane-rajasi — CHANGES_REQUESTED** [review]
One comment inline

File: src/core/tests/Test_StatusHandler.py (hunk 4 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py]: removed,
done

File: src/core/tests/Test_StatusHandler.py [hunk 5]
```diff
@@ -414,6 +414,14 @@ def test_sort_packages_by_classification_and_state(self):
             self.assertEqual(installation_patches_sorted[12]["name"], "test-package-2")  # | Other              | Excluded    |
             self.assertEqual(installation_patches_sorted[13]["name"], "test-package-1")  # | Other              | NotSelected |
 
+    def test_sequence_number_changed_termination_configure_patching_only(self):
+        """ Test sequence number change for configure patching throws newer operation superseded error message """
+        self.__assert_sequence_num_changed_termination(config=Constants.CONFIGURE_PATCHING, summary=Constants.CONFIGURE_PATCHING_SUMMARY, status=Constants.STATUS_ERROR)
+
+    def test_sequence_number_changed_termination_installation_patching(self):
```
Comment (rane-rajasi) [inline src/core/tests/Test_StatusHandler.py:421]: nit: rename to test_sequence_number_changed_termination_installation(self)

File: src/core/tests/Test_StatusHandler.py [hunk 6]
```diff
@@ -414,6 +414,14 @@ def test_sort_packages_by_classification_and_state(self):
...
+        """ Test sequence number change for installation patching throws newer operation superseded error message """
```
Comment (rane-rajasi) [inline src/core/tests/Test_StatusHandler.py:422]: nit: similar to the comment above, """ Test sequence number change for installation ..... """"

File: src/core/tests/Test_StatusHandler.py (hunk 5 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py]: done

**Review by kjohn-msft — APPROVED** [review]

**Review by rane-rajasi — APPROVED** [review]

**Pull request merged**